### ISSUE TITLE
Error when get-schema is applied to a schema with a type with only a collection relation

### ISSUE THREAD

**Issue by aqwert** [Urigo/graphql-cli#319]
graphql-cli v2.16.2

Schema example:

```
type Post {
   others: [Other!]!
}

type Other {
   id: ID! @unique
}
```

When using the `graphql get-schema --project database` command to create the generated schema it will display the following error

```
{ GraphQLError: Syntax Error: Expected Name, found }
    at syntaxError (/Users/.../projects/.../node_modules/graphql/error/syntaxError.js:24:10)
    at expect (/Users/.../projects/.../node_modules/graphql/language/parser.js:1299:32)
    at parseName (/Users/.../projects/.../node_modules/graphql/language/parser.js:92:15)
    at parseFieldDefinition (/Users/.../projects/.../node_modules/graphql/language/parser.js:809:14)
    at many (/Users/.../projects/.../node_modules/graphql/language/parser.js:1348:16)
    at parseFieldsDefinition (/Users/.../projects/.../node_modules/graphql/language/parser.js:799:50)
    at parseObjectTypeDefinition (/Users/.../projects/.../node_modules/graphql/language/parser.js:757:16)
    at parseTypeSystemDefinition (/Users/.../projects/.../node_modules/graphql/language/parser.js:663:16)
    at parseDefinition (/Users/.../projects/.../node_modules/graphql/language/parser.js:143:16)
    at parseDocument (/Users/.../projects/.../node_modules/graphql/language/parser.js:110:22)
  message: 'Syntax Error: Expected Name, found }',
  locations: [ { line: 193, column: 1 } ],
  path: undefined }
```
 This error goes away if both happen:
* a field is added, either an id: `id: ID! @unique` or other scalar field `name: String` to the type with only the collection (Post in this case)
* the generated files are deleted. The error remains if the file from a previous generation exists.

As a side issue with using:  `graphql get-schema --project database` and `graphql codegen` as part of the `prisma.yml`...
```
hooks:
  post-deploy:
    - graphql get-schema --project database
    - graphql codegen
```
The error is only displayed at the end of the codegen step, which initially made me think that was the issue. It was not since running the commands individually would result in the `get-schema` command to show the error.

**Comment by divyenduz** [comment]
Looks like this is a backend issue with invalid SDL being generated. Created a separate issue for that [here](https://github.com/prismagraphql/prisma/issues/2623).

However, we should not write the `prisma.graphql` file the first time when it is invalid.

@timsuchanek : Opinion?

**Comment by ardatan** [comment]
Currently we have new structure for GraphQL CLI and in the new alpha version `4.0.0-alpha.0` we suggest to use GraphQL Code Generator plugin of GraphQL CLI with Codegen's `schema-ast` plugin. Checkout new instructions and we will be happy to hear your feedback. Feel free to contact us using our Discord channel;
https://discord.gg/xud7bH

**Comment by Urigo** [comment]
We've just released a new alpha version channel for GraphQL CLI - `4.0.0-alpha.XXX`!

We've changed the `get-schema` command to use [GraphQL Code Generator](http://graphql-code-generator.com) plugin of GraphQL CLI with the Codegen's [`schema-ast` plugin](https://graphql-code-generator.com/docs/plugins/schema-ast) which can handle schema directives.

Checkout new instructions in the [migration guide](https://github.com/urigo/graphql-cli#migration-from-3xx-to-4xx) and let us know your feedback during that alpha period!

Feel free to contact us by opening a new issue or joining our Discord channel;
https://discord.gg/xud7bH
